engine/samples/voxel-shape-collisions/main.cpp

```diff
@@ -0,0 +1,223 @@
+#include <cubos/engine/collisions/shapes/capsule.hpp>
+#include <cubos/engine/collisions/shapes/voxel.hpp>
+#include <cubos/engine/defaults/plugin.hpp>
+#include <cubos/engine/fixed_step/plugin.hpp>
+#include <cubos/engine/gizmos/plugin.hpp>
+#include <cubos/engine/gizmos/target.hpp>
+#include <cubos/engine/physics/plugin.hpp>
+#include <cubos/engine/physics/solver/plugin.hpp>
+#include <cubos/engine/render/camera/camera.hpp>
+#include <cubos/engine/render/camera/draws_to.hpp>
+#include <cubos/engine/render/camera/perspective.hpp>
+#include <cubos/engine/render/defaults/plugin.hpp>
+#include <cubos/engine/render/defaults/target.hpp>
+#include <cubos/engine/render/picker/plugin.hpp>
+#include <cubos/engine/render/tone_mapping/plugin.hpp>
+#include <cubos/engine/render/voxels/grid.hpp>
+#include <cubos/engine/render/voxels/palette.hpp>
+#include <cubos/engine/settings/plugin.hpp>
+#include <cubos/engine/settings/settings.hpp>
+#include <cubos/engine/transform/plugin.hpp>
+#include <cubos/engine/voxels/plugin.hpp>
+#include <cubos/engine/window/plugin.hpp>
+
+using cubos::core::geom::Box;
+using cubos::core::io::Key;
+using cubos::core::io::Modifiers;
+using cubos::core::io::MouseButton;
+
+using namespace cubos::engine;
+
+static CUBOS_DEFINE_TAG(collisionsSampleUpdated);
+
+struct DebugDraw
+{
+    CUBOS_ANONYMOUS_REFLECT(DebugDraw);
+
+    bool normal = true;
+    bool points = true;
+    bool manifoldPolygon = true;
+};
+
+struct State
+{
+    CUBOS_ANONYMOUS_REFLECT(State);
+
+    bool collided = false;
+
+    Entity a;
+    Entity b;
+
+    glm::vec3 aRotationAxis;
+    glm::vec3 bRotationAxis;
+};
+
+/// [Get handles to assets]
+static const Asset<VoxelGrid> CarAsset = AnyAsset("059c16e7-a439-44c7-9bdc-6e069dba0c75");
+static const Asset<VoxelPalette> PaletteAsset = AnyAsset("1aa5e234-28cb-4386-99b4-39386b0fc215");
+/// [Get handles to assets]
+
+int main()
+{
+    auto cubos = Cubos();
+
+    cubos.plugin(defaultsPlugin);
+    cubos.tag(gizmosDrawTag).after(toneMappingTag);
+
+    cubos.resource<State>();
+    cubos.resource<DebugDraw>();
+
+    cubos.startupSystem("setup camera").call([](Commands commands) {
+        auto targetEnt = commands.create().add(RenderTargetDefaults{}).add(GizmosTarget{}).entity();
+        commands.create()
+            .relatedTo(targetEnt, DrawsTo{})
+            .add(Camera{.zNear = 0.1F, .zFar = 100.0F})
+            .add(PerspectiveCamera{.fovY = 60.0F})
+            .add(LocalToWorld{})
+            .add(Position{{-50.0F, 1.5F, 0.0F}})
+            .add(Rotation::lookingAt({3.0F, -1.0F, 0.0F}, glm::vec3{0.0F, 1.0F, 0.0F}));
+    });
+
+    cubos.startupSystem("configure Assets").tagged(settingsTag).call([](Settings& settings) {
+        settings.setString("assets.io.path", SAMPLE_ASSETS_FOLDER);
+    });
+
+    /// [Set palette]
+    cubos.startupSystem("set palette").call([](RenderPalette& palette) { palette.asset = PaletteAsset; });
+    /// [Set palette]
```

Comment:
```suggestion
    cubos.startupSystem("set palette").call([](RenderPalette& palette) { palette.asset = PaletteAsset; });
```
You aren't using these, since the sample doesn't have its own page anyway.